x-pack/plugins/enterprise_search/public/applications/app_search/components/analytics/components/analytics_header.tsx

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import { useValues } from 'kea';
+
+import queryString from 'query-string';
+import moment from 'moment';
+
+import { i18n } from '@kbn/i18n';
+import {
+  EuiPageHeader,
+  EuiPageHeaderSection,
+  EuiTitle,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSelect,
+  EuiDatePickerRange,
+  EuiDatePicker,
+  EuiButton,
+} from '@elastic/eui';
+
+import { KibanaLogic } from '../../../../shared/kibana';
+import { LogRetentionTooltip, LogRetentionOptions } from '../../log_retention';
+
+import { AnalyticsLogic } from '../';
+import { SERVER_DATE_FORMAT } from '../constants';
+import { convertTagsToSelectOptions } from '../utils';
+
+interface Props {
+  title: string;
+}
+export const AnalyticsHeader: React.FC<Props> = ({ title }) => {
+  const { allTags } = useValues(AnalyticsLogic);
+  const { history } = useValues(KibanaLogic);
+
+  // Parse out existing filters from URL query string
+  const { start, end, tag } = queryString.parse(history.location.search);
+  const [startDate, setStartDate] = useState(
+    start ? moment(start, SERVER_DATE_FORMAT) : moment().subtract(6, 'days')
+  );
+  const [endDate, setEndDate] = useState(end ? moment(end, SERVER_DATE_FORMAT) : moment());
+  const [currentTag, setCurrentTag] = useState((tag as string) || '');
```

Comment:
I don't think you need the cast here.

```suggestion
  const [currentTag, setCurrentTag] = useState(tag || '');
```